doc/ilc_app_interface.md

```diff
@@ -0,0 +1,104 @@
+# ILC to App interface
+
+This document aims to describe communication interface used by ILC
+to talk to apps (micro frontends).
+
+## Client side interface
+
+During the course of a single-spa page, registered applications are loaded, bootstrapped (initialized), mounted, unmounted, and unloaded.
+ILC (with the help of the [single-spa](https://single-spa.js.org/)) provides hooks into each phase via `lifecycles`.
+
+See more information about the [lifecycle functions here](https://single-spa.js.org/docs/building-applications/#lifecyle-props).
+
+TS interface for custom props that are passed to every lifecycle function 
+[can be found here](https://namecheap.github.io/ilc-sdk/interfaces/app.applifecyclefnprops.html).
+
+
+
+### Init code during app bundle loading
+
+Sometimes you need to run some initialization code right after app bundle will be loaded in the browser and usually you
+want to be able to pass some configuration properties to that code.
+
+ILC allows you to export a function called `mainSpa(props)` that will receive application properties that were defined in
```

Comment:
should we add information about default export?
https://github.com/namecheap/ilc/blob/master/ilc/client/BundleLoader.js#L73